src/Page.js

```diff
@@ -758,6 +765,39 @@ Page.prototype.insertTemporaryStyles = function (pageData) {
   return $.html();
 };
 
+Page.prototype.collectHeader = function () {
+  const $ = cheerio.load(this.content, { xmlMode: false });
+  const header = $('header');
+  if (header.length === 0) {
+    return;
+  }
+  this.headerHtml = htmlBeautify($.html('header'), { indent_size: 2 }).trim();
+  header.remove();
+  this.content = htmlBeautify($.html(), { indent_size: 2 });
+};
+
+Page.prototype.collectSiteNavigation = function () {
```

Comment:
This method seems rather similar to `collectHeader`. Would it make sense to abstract this behaviour into another method?

Reply:
Same applies for the `collectFooter` method.

Reply:
Now that you mentioned it, we could abstract it to 

```Javascript
Page.prototype.collectPageSection = function (section) {
  const $ = cheerio.load(this.content, { xmlMode: false });
  const selectedSection = $(section)
  ....
  this.pageSections[section] = $.html(section);
}

Page.collectAllPageSection = function () {
  this.collectPageSection('header');
  this.collectPageSection(`#${SITE_NAV_ID}`);
  this.collectPageSection('footer');
}
```